client/modules/IDE/components/Searchbar/Searchbar.jsx

```diff
@@ -1,82 +1,78 @@
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
 import { throttle } from 'lodash';
 import { withTranslation } from 'react-i18next';
 import i18next from 'i18next';
 import SearchIcon from '../../../../images/magnifyingglass.svg';
 
-class Searchbar extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      searchValue: this.props.searchTerm
-    };
-    this.throttledSearchChange = throttle(this.searchChange, 500);
-  }
+const Searchbar = ({
+  searchTerm,
+  setSearchTerm,
+  resetSearchTerm,
+  searchLabel,
+  t
+}) => {
+  const [searchValue, setSearchValue] = useState(searchTerm);
 
-  componentWillUnmount() {
-    this.props.resetSearchTerm();
-  }
+  const searchChange = () => {
+    setSearchTerm(searchValue.trim());
+  };
+
+  const throttledSearchChange = throttle(searchChange, 500);
```

Comment:
I have to load up your branch and make sure that this works as expected.  It seems like you would get a new value of `throttledSearchChange` on every re-render, which happens on every typed character, so I'm concerned that it wouldn't throttle properly.  It may need memoization.

Reply:
I think this would work.
const throttledSearchChange = useMemo(() => throttle(searchChange, 500), [searchChange]);

Reply:
shall i make a pr with this